js/modules/k6/data/share_test.go

```diff
@@ -0,0 +1,189 @@
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package data
+
+import (
+	"context"
+	"testing"
+
+	"github.com/dop251/goja"
+	"github.com/loadimpact/k6/js/common"
+	"github.com/stretchr/testify/require"
+)
+
+const makeArrayScript = `
+var array = new data.SharedArray("shared",function() {
+    var n = 50;
+    var arr = new Array(n);
+    for (var i = 0 ; i <n; i++) {
+        arr[i] = {value: "something" +i};
+    }
+	return arr;
+});
+`
+
+func newConfiguredRuntime(initEnv *common.InitEnvironment) (*goja.Runtime, error) {
+	rt := goja.New()
+	rt.SetFieldNameMapper(common.FieldNameMapper{})
+
+	ctx := common.WithInitEnv(context.Background(), initEnv)
+	ctx = common.WithRuntime(ctx, rt)
+	rt.Set("data", common.Bind(rt, new(data), &ctx))
+	_, err := rt.RunString("var SharedArray = data.SharedArray;")
+
+	return rt, err
+}
+
+func TestSharedArrayConstructorExceptions(t *testing.T) {
+	t.Parallel()
+	initEnv := &common.InitEnvironment{
+		SharedObjects: common.NewSharedObjects(),
+	}
+	rt, err := newConfiguredRuntime(initEnv)
+	require.NoError(t, err)
+	cases := map[string]struct {
+		code, err string
+	}{
+		"returning string": {
+			code: `new SharedArray("wat", function() {return "whatever"});`,
+			err:  "only arrays can be made into SharedArray",
+		},
+		"empty name": {
+			code: `new SharedArray("", function() {return []});`,
+			err:  "empty name provided to SharedArray's constructor",
+		},
+		"function in the data": {
+			code: `
+			var s = new SharedArray("wat2", function() {return [{s: function() {}}]});
+			if (s[0].s !== undefined) {
+				throw "s[0].s should be undefined"
+			}
+		`,
+			err: "",
+		},
+	}
+
+	for name, testCase := range cases {
+		name, testCase := name, testCase
+		t.Run(name, func(t *testing.T) {
+			_, err := common.RunString(rt, testCase.code)
+			if testCase.err == "" {
+				require.NoError(t, err)
+				return // the t.Run
+			}
+
+			require.Error(t, err)
+			exc := err.(*goja.Exception)
+			require.Contains(t, exc.Error(), testCase.err)
+		})
+	}
+}
+
+func TestSharedArrayAnotherRuntimeExceptions(t *testing.T) {
+	t.Parallel()
+
+	initEnv := &common.InitEnvironment{
+		SharedObjects: common.NewSharedObjects(),
+	}
+	rt, err := newConfiguredRuntime(initEnv)
+	require.NoError(t, err)
+	_, err = rt.RunString(makeArrayScript)
+	require.NoError(t, err)
+
+	// create another Runtime with new ctx but keep the initEnv
+	rt, err = newConfiguredRuntime(initEnv)
+	require.NoError(t, err)
+	_, err = rt.RunString(makeArrayScript)
+	require.NoError(t, err)
```

Comment:
Couldn't figure out what to name this block of code ... so I didn't move it to a function :shrug: